IdentityCore/src/network/error_handler/MSIDAADRequestErrorHandler.m

```diff
@@ -150,7 +150,13 @@ - (void)handleError:(NSError *)error
         }
     }
     
-    NSError *httpError = MSIDCreateError(MSIDHttpErrorCodeDomain, MSIDErrorServerUnhandledResponse, errorDescription, nil, nil, nil, context.correlationId, additionalInfo, YES);
+    NSError *httpUnderlyingError = nil;
+    if (httpResponse.statusCode == 403 || httpResponse.statusCode == 404)
+    {
+        httpUnderlyingError = MSIDCreateError(MSIDHttpErrorCodeDomain, MSIDErrorUnExpectedHttpResponse, errorDescription, nil, nil, nil, context.correlationId, nil, YES);
+    }
+
+    NSError *httpError = MSIDCreateError(MSIDHttpErrorCodeDomain, MSIDErrorServerUnhandledResponse, errorDescription, nil, nil, httpUnderlyingError, context.correlationId, additionalInfo, YES);
```

Comment:
Question, should the errorCode remain as `MSIDErrorServerUnhandledResponse`, since the `MSIDErrorUnexpectedHttpResponse` is now added as underlying error code?

Reply:
I didnt change it because we want to keep the top level error same.

Reply:
Oh yeah, you are right, I got confused.